### PR TITLE
(maint) Update to new signing key

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

**Comment by melissa** [comment]
cc/ @highb @ericwilliamson @haus you three have made a few recent commits to this repo. I want to check to ensure this isn't dead code and it worth updating. Thanks!

**Comment by haus** [comment]
@melissa deprecated, but still shipped in PE 3.8 afaik

**Comment by haus** [comment]
@melissa looks like the yum repo definition needs updating here too

**Comment by melissa** [comment]
Oops, that's for pointing that out! I'll get that updated now

**Comment by melissa** [comment]
Updated. PE 3.8 will continue to use the new key, but if we're shipping this module and expect it to work with the open source repos, we'll need this change

File: lib/puppet/cloudpack/scripts/master_source_dev.erb [hunk 1]
```diff
@@ -13,7 +13,7 @@ if [ -f /etc/redhat-release ]; then
 name=Puppet Labs PS (Unofficial)
 baseurl=http://yum.puppetlabs.com/el/\$releasever/dependencies/\$basearch
 enabled=1
-gpgkey=http://yum.puppetlabs.com/RPM-GPG-KEY-puppetlabs
+gpgkey=http://yum.puppetlabs.com/RPM-GPG-KEY-puppet
```
Comment (haus) [inline lib/puppet/cloudpack/scripts/master_source_dev.erb]: I think you'll want to add the new key and not replace it here (and elsewhere), as older versions of puppet will still be signed with the old key.

```gpgkey=http://yum.puppetlabs.com/RPM-GPG-KEY-puppetlabs
                http://yum.puppetlabs.com/RPM-GPG-KEY-puppet```

^^ is the right syntax according to https://projects.puppetlabs.com/issues/8867.

File: lib/puppet/cloudpack/scripts/puppet-community.erb [hunk 2]
```diff
@@ -43,7 +43,7 @@ function rpm_install() {
 
 function apt_install() {
   # Download and install the puppetlabs apt public
-  apt-key adv --recv-key --keyserver pool.sks-keyservers.net 4BD6EC30
+  apt-key adv --recv-key --keyserver pool.sks-keyservers.net 7F438280EF8D349F
```
Comment (haus) [inline lib/puppet/cloudpack/scripts/puppet-community.erb]: This keyserver seems to be toast afaict. Might be worth replacing with `pgp.mit.edu`?

**Comment by melissa** [comment]
Updated! I thought I had pushed the update out yesterday, but apparently I never did...

**Comment by ericwilliamson** [comment]
@melissa Just a reminder that this module is not shipped as part of the PE modules, but as part of the cloud provisioner package..I think

**Comment by melissa** [comment]
This code will still fail once we switch over to the new key, so it'll need these changes. The open source repos will all have the new key

File: lib/puppet/cloudpack/scripts/master_source_dev.erb [hunk 3]
```diff
@@ -14,6 +14,7 @@ name=Puppet Labs PS (Unofficial)
 baseurl=http://yum.puppetlabs.com/el/\$releasever/dependencies/\$basearch
 enabled=1
 gpgkey=http://yum.puppetlabs.com/RPM-GPG-KEY-puppetlabs
+       http://yum.puppetlabs.com/RPM-GPG-KEY-puppetlabs
```
Comment (shrug) [inline lib/puppet/cloudpack/scripts/master_source_dev.erb]: Looks like this duplicates the legacy key?

Comment (melissa) [inline lib/puppet/cloudpack/scripts/master_source_dev.erb]: womp womp. Fixed!

**Comment by shrug** [comment]
👍  from me. Those test failures don't look related...

File: lib/puppet/cloudpack/scripts/puppet-community.erb [hunk 4]
```diff
@@ -44,6 +47,7 @@ function rpm_install() {
 function apt_install() {
   # Download and install the puppetlabs apt public
   apt-key adv --recv-key --keyserver pool.sks-keyservers.net 4BD6EC30
```
Comment (haus) [inline lib/puppet/cloudpack/scripts/puppet-community.erb]: want to replace this with the long key id as long as you're in here?

Comment (melissa) [inline lib/puppet/cloudpack/scripts/puppet-community.erb]: Done

**Comment by haus** [comment]
gem versions are fucking travis up

**Comment by haus** [comment]
"Gem::InstallError: mime-types-data requires Ruby version >= 2.0.
An error occurred while installing mime-types-data (3.2016.0521), and Bundler" < from 1.9.3 cell

**Comment by melissa** [comment]
Travis failures are unrelated to these changes. Merge?

**Comment by haus** [comment]
I'm a fan of fixing broken shit when it gets seen, but how about at least opening a ticket for the travis failures?

File: Gemfile [hunk 5]
```diff
@@ -15,7 +15,7 @@ end
 
 gem 'fog', :require => false
 gem 'guid', :require => false
-gem 'google-api-client', :require => false
+gem 'google-api-client', "~> 0.8.7", :require => false
```
Comment (melissa) [inline Gemfile]: @haus fine, I fixed it.

Comment (melissa) [inline Gemfile]: Well, I fixed the gem version failure, but not the spec test failure.

On ruby 1.9.3:
```
/home/melissa/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/fog-profitbricks-2.0.1/lib/fog/profitbricks.rb:4:in `<top (required)>': undefined local variable or method `__dir__' for main:Object (NameError)
	from /home/melissa/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/fog-1.38.0/lib/fog.rb:51:in `require'
	from /home/melissa/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/fog-1.38.0/lib/fog.rb:51:in `<top (required)>'
	from /home/melissa/puppetlabs-cloud_provisioner/lib/puppet/cloudpack.rb:4:in `require'
...
```

On ruby 2.0.0:
```
/home/melissa/puppetlabs-cloud_provisioner/lib/puppet/cloudpack.rb:5:in `require': cannot load such file -- net/ssh (LoadError)
	from /home/melissa/puppetlabs-cloud_provisioner/lib/puppet/cloudpack.rb:5:in `<top (required)>'
	from /home/melissa/puppetlabs-cloud_provisioner/spec/spec_helper.rb:15:in `require'
	from /home/melissa/puppetlabs-cloud_provisioner/spec/spec_helper.rb:15:in `<top (required)>'
...
```

Do you want me to try to address those failures as well?

**Comment by haus** [comment]
@melissa the 1.9.3 failure looks like a problem in fog-profitbricks, which is being fixed (but looks unreleased) in https://github.com/fog/fog-profitbricks/commit/f5c25e38b43331f7f0771e24517681555f9583fc

**Comment by haus** [comment]
@melissa oh my god that is a sad road full of redness. I'm just gonna merge this. Getting tests to pass is a nightmare.

**Comment by melissa** [comment]
Thank you Haus. What are the next steps for this? Is there CI or should I make sure it's promoted into something?

**Comment by haus** [comment]
@melissa i have no idea. but it needs to be built for PE 3.8. probably via enterprise-dist rake tasks.

**Pull request merged**